modules/python/python-dest.c

```diff
@@ -178,7 +179,36 @@ _py_invoke_is_opened(PythonDestDriver *self)
 static gboolean
 _py_invoke_open(PythonDestDriver *self)
 {
-  return _dd_py_invoke_bool_method_by_name(self, "open");
+  if (!self->py.open)
+    return TRUE;
+
+  PyObject *ret;
+  gboolean result = FALSE;
+
+  ret = _py_invoke_function(self->py.open, NULL, self->class, self->super.super.super.id);
```

Comment:
You can use `_py_invoke_method_by_name`, so you don't have to "cache" this rarely-called method.

Reply:
I tried to do so, but it broke the code. The unfortunate thing is that, if there is no `open` method, then I want to return true. However, if exception happens during `open`, I want to return with false. In both cases, `py_invoke_method_by_name` returns NULL.

Adding an extra lookup for existence of `open` method might work. Based on that, I could early return with true.

Do you think that would be better than the caching version? If so, I can try to implement it.

Reply:
Oh okay, I understand it. The cache should be OK, if you prefer it this way.